lms/djangoapps/certificates/signals.py

```diff
@@ -79,6 +85,28 @@ def listen_for_passing_grade(sender, user, course_id, **kwargs):  # pylint: disa
 
     If needed, generate a certificate task.
     """
+    context = contexts.course_context_from_course_id(course_id)
```

Comment:
[general discussion] I'm curious about our usage of one vs many receivers for the completion signal. We have [another receiver](https://github.com/edx/edx-platform/blob/2220a1bcb209d288268856d1f8ef062f933b870f/openedx/features/enterprise_support/signals.py#L75-L86) that handles kicking off completion reporting to the integrated channels. Is there a downside of having a specific receiver meant to omit a segment event, or would we prefer to group this with the certificate generation?

Reply:
> We are going to emit this event course grade passed weather a certificate is enabled or not.

especially considering that we want this emitting even when a certificate isn't generated.

Reply:
We are emitting a general event for course completion.  Basically we want to emit this event on any course completion. In case of enterprise courses we just want to add enterprise uuid in the context just to filter out the events. Purpose of using certificates single to make sure course completion  event got trigger on all courses completion